src/main/java/com/emc/ecs/nfsclient/nfs/io/NfsFileInputStream.java

```diff
@@ -322,11 +325,15 @@ private void checkForClosed() throws IOException {
     }
 
     /**
-     * If the buffer has no more bytes to be read, load more bytes.
+     * If the buffer has no more bytes to be read, and there are bytes available in the file, load more bytes.
      * 
      * @throws IOException
      */
     private void loadBytesAsNeeded() throws IOException {
+        if (available() >= 0) {
```

Comment:
I'm confused by this conditional... it seems to contradict the method contract (in the javadoc). Should this be available() <= 0 ?  Otherwise, please add a line doc to explain